ros/Makefile

```diff
@@ -15,30 +22,44 @@ deps:
 
 .PHONY: build
 build:
-	colcon build
+	colcon build --cmake-args -DUSE_LOCAL_SDK=$(USE_LOCAL_SDK)
 
 .PHONY: test
 test:
 	colcon test
 	colcon test-result --all --verbose
 
-ROS_DISTRIBUTIONS := humble iron jazzy kilted rolling
-
 define generate_ros_targets
-.PHONY: docker-build-$(1)
-docker-build-$(1):
+.PHONY: docker-build-container-$(1)
+docker-build-container-$(1):
 	docker build -t foxglove-sdk-ros-$(1) -f ../Dockerfile.ros \
 		--build-arg ROS_DISTRIBUTION=$(1) \
+		$(DOCKER_USER_ARGS) \
 		$(EXTRA_DOCKER_ARGS) ..
 
+.PHONY: docker-build-$(1)
+docker-build-$(1): docker-build-container-$(1)
+	docker run --rm -v $(PWD)/..:/sdk foxglove-sdk-ros-$(1) make USE_LOCAL_SDK=$(USE_LOCAL_SDK) build
+
+.PHONY: docker-test-$(1)
+docker-test-$(1):
+	docker run --rm -v $(PWD)/..:/sdk foxglove-sdk-ros-$(1) make test
+
 .PHONY: docker-sh-$(1)
 docker-sh-$(1):
 	docker run --rm -it -v $(PWD)/..:/sdk foxglove-sdk-ros-$(1)
+
 endef
 $(foreach distribution,$(ROS_DISTRIBUTIONS),$(eval $(call generate_ros_targets,$(strip $(distribution)))))
 
+.PHONY: docker-build-container
+docker-build-container: docker-build-container-rolling
```

Comment:
nit: I would probably call this `docker-build-image` or just `docker-image`

technically a container is an instance of an image